src/super_gradients/training/utils/quantization_utils.py

```diff
@@ -0,0 +1,384 @@
+
+(Licensed under the Apache License, Version 2.0)
+"""
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+import torch
+from super_gradients.common.abstractions.abstract_logger import get_logger
+from super_gradients.training.utils.callbacks import Phase, PhaseCallback, PhaseContext
+import os
+from enum import Enum
+from super_gradients.training.utils.checkpoint_utils import load_checkpoint_to_model
+from super_gradients.training.utils import get_param
+from super_gradients.training.utils.distributed_training_utils import get_local_rank, \
+    get_world_size
+from torch.distributed import all_gather
+
+logger = get_logger(__name__)
+
+try:
+    _imported_trt_failure = None
+except (ImportError, NameError, ModuleNotFoundError) as import_err:
+    logger.warning("Failed to import pytorch_quantization")
+    _imported_trt_failure = import_err
+
+try:
+    from pytorch_quantization import nn as quant_nn, quant_modules
+    from pytorch_quantization import calib
+    from pytorch_quantization.tensor_quant import QuantDescriptor
+
+    _imported_pytorch_quantization_failure = None
+except (ImportError, NameError, ModuleNotFoundError) as import_err:
+    logger.warning("Failed to import pytorch_quantization")
+    _imported_pytorch_quantization_failure = import_err
+
+
+class QuantizationLevel(str, Enum):
+    FP32 = 'FP32'
+    FP16 = 'FP16'
+    INT8 = 'INT8'
+    HYBRID = 'Hybrid'
+
+    @staticmethod
+    def from_string(quantization_level: str) -> Enum:
+        quantization_level = quantization_level.lower()
+        if quantization_level == 'fp32':
+            return QuantizationLevel.FP32
+        elif quantization_level == 'fp16':
+            return QuantizationLevel.FP16
+        elif quantization_level == 'int8':
+            return QuantizationLevel.INT8
+        elif quantization_level == 'hybrid':
+            return QuantizationLevel.HYBRID
+        else:
+            raise NotImplementedError(f'Quantization Level: "{quantization_level}" is not supported')
+
+
+def export_qat_onnx(model: torch.nn.Module, onnx_filename: str, input_shape: tuple,
+                    per_channel_quantization: bool = False):
+    """
+    Method for exporting onnx after QAT.
+
+    :param model: torch.nn.Module, model to export
+    :param onnx_filename: str, target path for the onnx file,
+    :param input_shape: tuple, input shape (usually BCHW)
+    """
+    if _imported_pytorch_quantization_failure is not None:
+        raise _imported_pytorch_quantization_failure
+    else:
+        model.eval()
+        quant_nn.TensorQuantizer.use_fb_fake_quant = True
+        # Export ONNX for multiple batch sizes
+        logger.info("Creating ONNX file: " + onnx_filename)
+        dummy_input = torch.randn(input_shape, device='cuda')
+        opset_version = 13 if per_channel_quantization else 12
+        torch.onnx.export(model, dummy_input, onnx_filename, verbose=False, opset_version=opset_version,
```

Comment:
what about prep_for_conversion?

Reply:
will this work with RepVGG block?

Reply:
I added the call in https://github.com/Deci-AI/super-gradients/pull/232/commits/8b23e100a550483a986df88171113fd6e2bfdd7f .

Regarding the RepVGG block- you are right, it probably won't work..
We need to see how we can handle the fusing with quantised layers, and perhaps have some arg passed to prep_model_for_conversion so we know to operate accordingly...